tracer/src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -55,6 +55,12 @@ public void Close(Scope scope)
 
             // scope.Parent is null for distributed traces, so use scope.Span.Context.Parent
             DistributedTracer.Instance.SetSpanContext(scope.Span.Context.Parent as SpanContext);
+
+            // Propagate the resource name to the profiler for root web spans
+            if (scope.Parent == null && scope.Span.Type == SpanTypes.Web)
```

Comment:
Do you definitely only want `SpanTypes.Web` here? What about `SpanTypes.Grpc`? or `SpanTypes.Serverless`? Why not _any_ root span? 🤔 Also, this doesn't _necessarily_ mean the span is closed right? Are we ok with that?

Reply:
I tend to agree, but for now the specification explicitly limits the feature to web spans.

> In order for this feature to work profiling needs to capture the ‘resource’ field of the ‘local root span’ of each request. For now we are focusing only on spans with type ‘WEB’.

Reply:
> specification explicitly limits the feature to web spans

You may want to add a code comment about this since you've already been asked twice :)

Reply:
Is the intention here to call `Profiler.Instance.ContextTracker.SetEndpoint()` whenever a trace finishes or when the root span is closed or something else? I feel like any tracking of this kind should happen in `TraceContext` because that is the class responsible for tracking a single trace's spans, including its root span. `ScopeManager` is only responsible for allowing access to the currently active span at a given point in time.

Reply:
>  I feel like any tracking of this kind should happen in TraceContext because that is the class responsible for tracking a single trace's spans, including its root span. ScopeManager is only responsible for allowing access to the currently active span at a given point in time.

I agree, that's what I was trying to get at (less eloquently) with: "Also, this doesn't necessarily mean the span is closed right? Are we ok with that?"

Reply:
+1 that `TraceContext` seems the most appropriate place to have this logic

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3015/commits/67afcdb49a6cb68e350088ddfe86bfedc6910915